### ISSUE TITLE
Block comments cause syntax error where regular comments don't

### ISSUE THREAD

**Issue by AGhost-7** [jashkenas/coffeescript#3993]
This will compile and run fine.
```coffeescript
router
# small description...
.get('/foo', (req, res, next) ->
)
```
This however, does not.
```coffeescript
router
###
  A longer comment...
###
.get('/foo', (req, res, next) ->
)
```
My expectation was that block comments should work the same as regular comments.

btw, version is 1.9.1 if that helps at all.

**Comment by michaelficarra** [comment]
Duplicate of many other duplicates.

**Comment by AGhost-7** [comment]
I did try to search for similar issues before I posted, and wasn't able to find anything. Could you please point me to where the original issue is?

**Comment by michaelficarra** [comment]
See https://github.com/jashkenas/coffeescript/issues/3717#issuecomment-62399306.

And there's been even more between then and now.

**Comment by AGhost-7** [comment]
Ok thanks, I just wanted to know why its not going to be fixed. I guess block comments are just weird like that, since they need to be parsed. Is this a problem in redux?

**Comment by vendethiel** [comment]
They just get dropped in CSR instead, [it seems](http://michaelficarra.github.io/CoffeeScriptRedux/#try:router%0D%0A%23%23%23%0D%0A%20%20A%20longer%20comment...%0D%0A%23%23%23%0D%0A.get('%2Ffoo'%2C%20(req%2C%20res%2C%20next)%20-%3E%0D%0A)).

**Comment by lydell** [comment]
The only times I’ve ever used block comments is at the top of files to make copyright and license info appear in the compiled JS. At all other times I recommend singleline comments.

**Comment by vendethiel** [comment]
They can be very useful to comment large blocks of code, imho. Outside from that...

**Comment by AGhost-7** [comment]
It doesn't parse comments at all? Hmm, I guess that would make sense. You shouldn't read the js output of the compiler imo, so I think thats a pretty good decision.

**Comment by lydell** [comment]
@vendethiel Don’t take us into the debate between what text editors should help you with, and what languages themselves should help you with ;P

@AGhost-7 No, it _does_ parse comments. It just does not compile them. And, in fact, one of CoffeeScript’s design goals is to have as readable output as possible so it is actually meant to be read.

**Comment by vendethiel** [comment]
> @vendethiel Don’t take us into the debate between what text editors should help you with, and what languages themselves should help you with ;P

Right, sorry :).

(another argument people use is for tools that embed metadata in comments, i.e. closure compiler)

**Comment by AGhost-7** [comment]
Might be the current design goal here, but as for me I think making preprocessor output readable is kind of a waste of time considering you have source maps. I just don't see the point, to be perfectly honest. CS imo is also pretty easy to pick up too, took me about a day to get comfortable, so I'm not sure if having that for developers who can't read CS is really essential in the end.

But I digress, thanks for the info.

**Comment by GeoffreyBooth** [comment]
This is fixed via #4572.
